CMakeLists.txt

```diff
@@ -711,6 +711,11 @@ function(build_tests)
       configure_file(EXAMPLES/MATRIX_MARKET/issue215.mtx ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/issue215.mtx)
       configure_file(EXAMPLES/MATRIX_MARKET/issue215.sh ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/issue215.sh)
       add_test(NAME issue215_tst WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY} COMMAND ${BASH_PROGRAM} -eu issue215.sh)
+      if (WIN32 AND BUILD_SHARED_LIBS)
+        set_tests_properties(arpackmm_tst issue401_tst issue215_tst
+          PROPERTIES
+          ENVIRONMENT_MODIFICATION "PATH=path_list_prepend:$<TARGET_FILE_DIR:arpack>")
```

Comment:
Shouldn't be this done for all tests? Why only theses 3 ones?

Reply:
Good point. I forgot to mention the crucial point in the PR description: `.dll` are also loaded from the current working directory on Windows (a security nightmare!).

Afaict, these three tests are the only ones that set the working directory to somewhere other than the directory with the libarpack.dll.

I'll update the description of the PR accordingly.

Reply:
> these three tests are the only ones that set the working directory to somewhere other than

These 3 tests seem to be the only ones to be ran from a sh script with modified WORKING_DIRECTORY: https://github.com/opencollab/arpack-ng/blob/77ba50e8007695af4299dc9709da709e48288eca/CMakeLists.txt#L706-L713

May be changing `add_test` or/and `configure_file` could be enough?

Does autotools have this problem (we try as much as possible to make the 2 cmake/autotools work the same way)?

Reply:
> 3 tests seem to be the only ones to be ran from a sh script with modified WORKING_DIRECTORY

WORKING_DIRECTORY could maybe simply be suppressed from `add_test` for these 3 tests: did you try? This way there would be nothing specific to any tests

Reply:
A possible alternative could be to put everything for these tests in the `CMAKE_BINARY_DIR` (i.e., the directory with the built libarpack). That is at least the generated .mtx files and the arpackmm binary. (The generated shell script could probably stay in a dedicated directory.)
The shell script tries to run the arpackmm binary in the current working directory. And iiuc, the arpackmm binary is looking for the .mtx files in the current working directory, too.

But that seems more messy than the current setup imho. Currently, the files for the Eigen tests are nicely separated in their own directory. Please, let me know if you prefer to change that anyway.

I haven't checked running the tests with the autotools build system on Windows yet. I'll try that after we know *how* this issue should be resolved.

Reply:
> A possible alternative could be to put everything for these tests in the CMAKE_BINARY_DIR

Not sure what to do here. Both solution look similar to me. @sylvestre: what do you say?